pkg/errinsrc/internal/golocation.go

```diff
@@ -103,7 +114,12 @@ func convertSingleGoPositionToRange(filename string, fileBody []byte, start toke
 	file, err := parser.ParseFile(fs, filename, fileBody, parser.ParseComments)
 
 	if err != nil || file == nil {
-		return start
+		fileBodyStr := string(fileBody)
```

Comment:
I just moved GuessEndColumn to be able to reuse it here. The original implementation was on strings for some reason: 

```
// Try and guess the atom where the error is
				// if the currentCause.Start()/currentCause.End() point is the same position
				if endCol <= startCol {
					endCol = GuessEndColumn(sc.Text(), startCol)
				}
```

Didn't want to change it in case it was intentional.

Reply:
Doubt it's intentional. Not the end of the world since we don't render errors that frequently, but might be nice to change if it's easy